quint/src/names/collector.ts

```diff
@@ -143,8 +148,6 @@ export class NameCollector implements IRVisitor {
         return
       }
 
-      // Update the definition to point to the expression being overriden
-      constDef.id = ex.id
```

Comment:
This was a kind of a hack so the compiler would evaluate the constant as the expression it was overridden to. We now do it on the level of registers on the evaluator.